src/main/java/net/snowflake/ingest/streaming/internal/ClientBufferParameters.java

```diff
@@ -0,0 +1,51 @@
+package net.snowflake.ingest.streaming.internal;
+
+import net.snowflake.ingest.utils.ParameterProvider;
+
+/** Channel's buffer relevant parameters that are set at the owning client level. */
+public class ClientBufferParameters {
+  private SnowflakeStreamingIngestClientInternal clientInternal;
+
+  private boolean enableParquetInternalBuffering;
+
+  private long maxChunkSizeInBytes;
+
+  private long maxAllowedRowSizeInBytes;
+
+  public ClientBufferParameters(
```

Comment:
Why do we need this constructor? Could you make it clear if this is test only?

Reply:
Right, this is test-only. I added a comment.